rfcs/006-regrading.md

```diff
@@ -0,0 +1,39 @@
+# Regrading
+
+# Summary
+
+Provide instructors with the options of regrading a question for a perticular student or whole class.
+
+# Motivation
+
+Sometime the internal/external grader for a question in homework or exam might not be set up correctly before the question is released to the student. Currently, if an instructor realize there was an issue with the grader of a question, the options the instructor got is very limited according to this [doc](../docs/regrading.md). We hope to give the instructor a wider range of options to resolve a broken question.
+
+Goals:
+* Instructor can go to the assessment instantce page to regrade a question for a particular student.
+* Instructor can go to the assessment questions page to regrade a question for the whole class.
+* The regrading job should take all the user submissions and rerun grader on all of them in the order as they come in to calculate the new score.
+* The instructor can choose between whether to replace the old score with a lower new score or not. 
+
+# Proposed solution
+
+## Backend
+
+### Overview
+We can modify the current [regrading](../lib/regrading.js) system and add the option to rerun the internal/external grader. The code in the backend that runs the grader can be found in [question.js](../lib/question.js). We can also add a new method in this file to handle the regrade functionality.
```

Comment:
From an instructor's perspective, the "regrade" means that the server.py grade(data) function is run on all submitted answers. Setting data["correct_answers"] occurs in generate(data), so "regrade" does *not* mean resetting data["correct_answers"], nor making changes to html. I think we should make grade(data) templates for fixing MC and checkbox questions.

Reply:
As an alternative for MC/checkbox questions, we could consider re-reading the `correct` status for each answer from `question.html` in the `grade()` function. At the moment we store this information in `params` during the `generate()` phase, which is why instructors can't change it later.

The main issue with re-reading `correct` status from the HTML during grading is how to match answers from `params` to those in the HTML. We could do this by index (answer-1, answer-2, etc) or by fuzzy-string match?